client/src/components/Ticketing/checkout/CheckoutPage.tsx

```diff
@@ -38,6 +39,14 @@ export default function CheckoutPage(): ReactElement {
   const [checkoutStep, setCheckoutStep] = useState<'donation' | 'form'>(
     'donation',
   );
+  const [popUp, setPopUp] = useState({
+    show: false,
+    title: '',
+    message: '',
+    success: true,
+    handleProceed: () => setPopUp((popUp) => ({...popUp, show: false})),
+    showSecondary: false,
+  });
```

Comment:
I noticed that the `x` button of the PopUp doesn't work, its likely that you'll need to add a `handleClose: () => setPopUp((popUp) => ({...popUp, show: false})),`, and with this observation, we probably don't want the `handleClose` function to be optional in the `PopUp` file. That ones on me